### DISCUSSION TITLE
How to add index on Repeater itemLabel ?

### DISCUSSION THREAD

**Discussion by AFSDARIF** [filamentphp/filament#8565] [Help] [answered]
![image](https://github.com/filamentphp/filament/assets/10229660/27e352b9-275d-42e1-85dd-c32e4f6e6678)

https://filamentphp.com/docs/3.x/forms/fields/repeater#adding-a-label-to-repeater-items-based-on-their-content
` ->itemLabel(fn (array $state): ?string => $state['name'] ?? null)`

**Comment by leandrocfe** [comment, marked as answer]
This code could be improved

```php
->itemLabel(function (array $state, $component): ?string {

    if (!$state['name']) {
        return null;
    }

    $key = array_search($state, $component->getState());
    $index = array_search($key, array_keys($component->getState()));

    return $index + 1;
})
```

**Reply by AFSDARIF** [reply]
Thanks, its work for me.
